sdk/pinocchio/src/entrypoint/lazy.rs

```diff
@@ -241,10 +233,37 @@ impl InstructionContext {
     /// before reading all accounts will result in undefined behavior.
     #[inline(always)]
     pub unsafe fn program_id_unchecked(&self) -> &Pubkey {
-        let data_len = *(self.input.add(self.offset) as *const usize);
-        &*(self
-            .input
-            .add(self.offset + core::mem::size_of::<u64>() + data_len) as *const Pubkey)
+        let data_len = *(self.input as *const usize);
+        &*(self.input.add(core::mem::size_of::<u64>() + data_len) as *const Pubkey)
+    }
+
+    /// Read an account from the input buffer.
+    ///
+    /// This can only be called with a buffer that was serialized by the runtime as
+    /// it assumes a specific memory layout.
+    #[allow(clippy::cast_ptr_alignment, clippy::missing_safety_doc)]
```

Comment:
Why disable `clippy::missing_safety_doc` when the comment already has the necessary documentation?

```suggestion
    /// Read an account from the input buffer.
    ///
    /// # Safety
    ///
    /// This can only be called with a buffer that was serialized by the runtime as
    /// it assumes a specific memory layout.
    #[allow(clippy::cast_ptr_alignment)]
```

Reply:
Good call, I will address this and the imports in a separate PR.